devicetypes/tests.py

```diff
@@ -21,6 +21,72 @@ def test_type_creation(self):
         self.assertEqual(mytype.get_absolute_url(), reverse('type-detail', kwargs={'pk': mytype.pk}) )
         self.assertEqual(mytype.get_edit_url(), reverse('type-edit', kwargs={'pk': mytype.pk}) )
 
+    def test_type_list(self):
+        '''method for testing the presentation and reachability of the list of devicestypes over several pages'''
+        devicetypes = mommy.make(Type, _quantity=40)
+
+        # testing if loading of devicetype-list-page was successful (statuscode 2xx)
+        url = reverse("type-list")
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, 200)
+
+        # testing the presentation of only 30 results of query on one page
+        self.assertEqual(len(resp.context["type_list"]), 30)
+        self.assertEqual(resp.context["paginator"].num_pages, 2)
+
+        # testing the successful loading of second page of devicetype-list (statuscode 2xx)
+        url = reverse("type-list", kwargs={"page": 2})
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, 200)
+
+    def test_type_detail(self):
+        '''method for testing the reachability of existing devicetypes'''
+        # querying all devices and choose first one to test
+        devicetype = mommy.make(Type)
+        devicetypes = Type.objects.all()
+        devicetype = devicetypes[0]
+
+        # test successful loading of detail-view of chossen devicetype (first one, statuscode 2xx)
+        url = reverse("type-detail", kwargs={"pk": devicetype.pk})
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, 200)
+
+    def test_type_add(self):
+        '''method for testing adding a devicetype'''
+        devicetype = mommy.make(Type)
+
+        # testing successful loading of devicetype-page of added device (statuscode 2xx)
```

Comment:
not quite true, see comment above.

This just tests if the type-add form view loads correctly.